main.go

```diff
@@ -732,9 +734,37 @@ func executeLinter(state *linterState, args []string) error {
 			state.Name)
 	}
 
-	processOutput(state, buf.Bytes())
+	cmdOutput, err := preprocessLinterPanic(state.Name, buf, err)
+	processOutput(state, cmdOutput)
 	debug("%s linter took %s", state.Name, time.Since(start))
-	return nil
+	return err
+}
+
+const panicExitStatus = 2
+
+func preprocessLinterPanic(name string, buf *bytes.Buffer, processError error) ([]byte, error) {
+	if exitStatus(processError) == panicExitStatus {
```

Comment:
Just to clarify, if the function returns then all the output from the linter would be discarded and gometalinter will think it panicked. 

It may not have panicked, so could we still send output to processOutput function but also log the warning? Then a false positive should just have an extra warning instead of discarding output and logging warning.